### PR TITLE
cw721-fixed-price: Fix mint message sent to wrong contract address

### PR DESCRIPTION
**Pull request by axvn**
The mint message on receive cw20 token was sent to the cw20 address instead of the cw721 address.

### CODE REVIEW

**Comment by axvn** [comment]
https://github.com/CosmWasm/cw-nfts/pull/48 would address the failed `wasm-build`

**Review by chingfeng2021 — DISMISSED** [review]

**Review by yuzhouzhouba — DISMISSED** [review]

File: contracts/cw721-fixed-price/src/contract.rs [hunk 1]
```diff
@@ -161,7 +161,7 @@ pub fn execute_receive(
     });
 
     let callback = CosmosMsg::Wasm(WasmMsg::Execute {
-        contract_addr: config.cw20_address.to_string(),
+        contract_addr: config.cw721_address.clone().unwrap().to_string(),
```
Comment (JakeHartnell) [inline contracts/cw721-fixed-price/src/contract.rs]: Can we do this without an `unwrap()` please? :pray:

Comment (axvn) [reply contracts/cw721-fixed-price/src/contract.rs]: Thanks for the review, it indeed might panic

**Review by JakeHartnell — DISMISSED** [review]
Much better without the `unwrap()`

**Review by the-frey — DISMISSED** [review]

File: contracts/cw721-fixed-price/src/contract.rs [hunk 2]
```diff
@@ -160,16 +160,19 @@ pub fn execute_receive(
         extension: config.extension.clone(),
     });
 
-    let callback = CosmosMsg::Wasm(WasmMsg::Execute {
-        contract_addr: config.cw20_address.to_string(),
-        msg: to_binary(&mint_msg)?,
-        funds: vec![],
-    });
+    if let Some(cw721) = &config.cw721_address {
+        let callback = CosmosMsg::Wasm(WasmMsg::Execute {
```
Comment (orkunkl) [inline contracts/cw721-fixed-price/src/contract.rs]: you can use `cw721-base/helper.rs` here

File: contracts/cw721-fixed-price/src/contract.rs [hunk 3]
```diff
@@ -160,16 +160,19 @@ pub fn execute_receive(
         extension: config.extension.clone(),
     });
 
-    let callback = CosmosMsg::Wasm(WasmMsg::Execute {
-        contract_addr: config.cw20_address.to_string(),
-        msg: to_binary(&mint_msg)?,
-        funds: vec![],
-    });
+    if let Some(cw721) = &config.cw721_address {
```
Comment (orkunkl) [inline contracts/cw721-fixed-price/src/contract.rs]: You can use `.ok_or(Err)` here to simplify code

File: contracts/cw721-fixed-price/src/contract.rs (hunk 2 above)
Comment (axvn) [reply contracts/cw721-fixed-price/src/contract.rs]: Solved, thanks for the hint.

File: contracts/cw721-fixed-price/src/contract.rs (hunk 3 above)
Comment (axvn) [reply contracts/cw721-fixed-price/src/contract.rs]: Had a try, not sure if it's more idiomatic / cleaner, I'll leave it as it is here.

**Review by JakeHartnell — DISMISSED** [review]

File: contracts/cw721-fixed-price/src/contract.rs [hunk 4]
```diff
@@ -160,23 +160,23 @@ pub fn execute_receive(
...
+    Err(ContractError::Cw721NotLinked {})
 }
 
 #[cfg(test)]
 mod tests {
     use super::*;
     use cosmwasm_std::testing::{mock_dependencies, mock_env, mock_info, MOCK_CONTRACT_ADDR};
-    use cosmwasm_std::{from_binary, to_binary, SubMsgExecutionResponse, SubMsgResult};
+    use cosmwasm_std::{
+        from_binary, to_binary, ContractResult, CosmosMsg, SubMsgExecutionResponse,
```
Comment (JakeHartnell) [inline contracts/cw721-fixed-price/src/contract.rs]:
```suggestion
        from_binary, to_binary, CosmosMsg, SubMsgExecutionResponse, SubMsgResult
```

**Review by JakeHartnell — DISMISSED** [review]

File: contracts/cw721-fixed-price/src/contract.rs [hunk 5]
```diff
@@ -160,23 +160,21 @@ pub fn execute_receive(
...
-    CONFIG.save(deps.storage, &config)?;
+    if let Some(cw721) = config.cw721_address.clone() {
+        let callback = Cw721Contract(cw721).call(mint_msg)?;
+        config.unused_token_id += 1;
+        CONFIG.save(deps.storage, &config)?;
 
-    Ok(Response::new().add_message(callback))
+        return Ok(Response::new().add_message(callback));
+    }
+    Err(ContractError::Cw721NotLinked {})
```
Comment (the-frey) [inline contracts/cw721-fixed-price/src/contract.rs]: Looks good, but tiny nit - shouldn't this block more idiomatically just be a match?

Comment (JakeHartnell) [reply contracts/cw721-fixed-price/src/contract.rs]: Yeah, that would be nicer.

File: contracts/cw721-fixed-price/src/contract.rs [hunk 6]
```diff
@@ -372,7 +370,7 @@ mod tests {
             res.messages[0],
             SubMsg {
                 msg: CosmosMsg::Wasm(WasmMsg::Execute {
-                    contract_addr: String::from(MOCK_CONTRACT_ADDR),
+                    contract_addr: "nftcontract".to_string(),
```
Comment (orkunkl) [inline contracts/cw721-fixed-price/src/contract.rs]: best to keep using  MOCK_CONTRACT_ADDR as a variable

Comment (JakeHartnell) [reply contracts/cw721-fixed-price/src/contract.rs]: Ah, but that's used for the cw20 address...

Comment (JakeHartnell) [reply contracts/cw721-fixed-price/src/contract.rs]: Added a constant for this, should be good.

**Review by JakeHartnell — APPROVED** [review]

**Review by Callum-A — APPROVED** [review]

**Pull request merged**